bluesky_adaptive/agents/base.py

```diff
@@ -565,9 +566,16 @@ def _check_queue_and_start(self):
             logger.info("Agent is starting an idle queue with exactly 1 item.")
 
     def add_suggestions_to_queue(self, batch_size: int):
-        """Calls ask, adds suggestions to queue, and writes out event"""
-        doc, next_points = self.ask(batch_size)
-        uid = self._write_event("ask", doc)
+        """Calls ask, adds suggestions to queue, and writes out events.
+        This will create one event for each suggestion.
+        """
+        docs, next_points = self.ask(batch_size)
+        uid = str(uuid.uuid4())
+        for batch_idx, (doc, next_point) in enumerate(zip(docs, next_points)):
+            doc["suggestion"] = next_point
+            doc["batch_idx"] = batch_idx
+            doc["batch_size"] = len(next_points)
+            self._write_event("ask", doc, uid=uid)
```

Comment:
This should be part of the data not the uid as there is an assumption that the event uids are unique.

Reply:
Fixed in [8705064](https://github.com/bluesky/bluesky-adaptive/pull/16/commits/8705064ef6d050ad639d468510c06c8174433205)